autoscrub/version_check.py

```diff
@@ -0,0 +1,53 @@
+import six
+import re
+from distutils.version import LooseVersion
+
+if six.PY3:
+    from subprocess import Popen, PIPE
+else:
+    # backported subprocess module
+    from subprocess32 import Popen, PIPE
+__terminal_encoding = "utf-8"
+
+
+def get_ffmpeg_version(strip_prefix=True):
+    """Gets the version of ffmpeg, stripping prefix comprising alpha/dash
+    characters, e.g. nightly build 'N-' prefix.
+    Returns None if version cannot be determined."""
+    command = ["ffmpeg", "-version"]
+    p = Popen(command, stdout=PIPE, stderr=PIPE)
+    stdout = p.stdout.readlines()
```

Comment:
Unfortunately we can't use `autoscrub._agnostic_communicate` (because that assumes ffmpeg is printing to stderr) but we should probably at least use `p.communicate()` here for reasons outlined [here](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.stdin). If using `_agnostic_Popen` above, we should include the rest of `_agnostic_communicate` from [this](https://github.com/philipstarkey/autoscrub/blob/a50c298b7b27c6a663620a2c62582eed4a17954c/autoscrub/__init__.py#L198) line onwards (for cleanup and error handling) which requires referencing the process list as `autoscrub._process_list` and exception as `autoscrub.AutoscrubException` assuming we add `import autoscrub` to the  file.

Reply:
I'd already tried `autoscrub._agnostic_communicate` and had arrived at the same conclusion. One option would be to not limit `autoscrub._agnostic_communicate` to stderr. The meaning of 'agnostic' in reference to the word of caution about [`Popen.stdin`](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.stdin) are lost on me right now (I've tested this on Python 2 and 3).